protopipe/pipeline/temp.py

```diff
@@ -26,6 +32,171 @@
 logger = logging.getLogger(__name__)
 
 
+def apply_simtel_r1_calibration(r0_waveforms,
+                                pedestal,
+                                dc_to_pe,
+                                gain_selector,
+                                calib_scale=1.0,
+                                calib_shift=0.0):
+    """
+    Perform the R1 calibration for R0 simtel waveforms. This includes:
+        - Gain selection
+        - Pedestal subtraction
+        - Conversion of samples into units proportional to photoelectrons
+          (If the full signal in the waveform was integrated, then the resulting
+          value would be in photoelectrons.)
+          (Also applies flat-fielding)
+    Parameters
+    ----------
+    r0_waveforms : ndarray
+        Raw ADC waveforms from a simtel file. All gain channels available.
+        Shape: (n_channels, n_pixels, n_samples)
+    pedestal : ndarray
+        Pedestal stored in the simtel file for each gain channel
+        Shape: (n_channels, n_pixels)
+    dc_to_pe : ndarray
+        Conversion factor between R0 waveform samples and ~p.e., stored in the
+        simtel file for each gain channel
+        Shape: (n_channels, n_pixels)
+    gain_selector : ctapipe.calib.camera.gainselection.GainSelector
+    calib_scale : float
+        Extra global scale factor for calibration.
+        Conversion factor to transform the integrated charges
+        (in ADC counts) into number of photoelectrons on top of dc_to_pe.
+        Defaults to no scaling.
+    calib_shift: float
+        Shift the resulting R1 output in p.e. for simulating miscalibration.
+        Defaults to no shift.
+    Returns
+    -------
+    r1_waveforms : ndarray
+        Calibrated waveforms
+        Shape: (n_pixels, n_samples)
+    selected_gain_channel : ndarray
+        The gain channel selected for each pixel
+        Shape: (n_pixels)
+    """
+    n_channels, n_pixels, n_samples = r0_waveforms.shape
+    ped = pedestal[..., np.newaxis]
+    DC_to_PHE = dc_to_pe[..., np.newaxis]
+    gain = DC_to_PHE * calib_scale
+    r1_waveforms = (r0_waveforms - ped) * gain + calib_shift
+    if n_channels == 1:
+        selected_gain_channel = np.zeros(n_pixels, dtype=np.int8)
+        r1_waveforms = r1_waveforms[0]
+    else:
+        selected_gain_channel = gain_selector(r0_waveforms)
+        r1_waveforms = r1_waveforms[selected_gain_channel, np.arange(n_pixels)]
+    return r1_waveforms, selected_gain_channel
+
+
+class MySimTelEventSource(SimTelEventSource):
```

Comment:
I guess you can remove this class now that it is in the master?

Reply:
the master branch of ctapipe? I cannot use the development version here, I use always the latest release (which currently is 0.11.0)

Reply:
that is the reason why I added the `protopipe.pipeline.temp` module in the first place

Reply:
ah, sorry, I thought this was tied to the master.  Ok then.